src/geom/sides.rs

```diff
@@ -266,3 +371,25 @@ impl<T: Fold> Fold for Sides<Option<T>> {
         })
     }
 }
+
+impl Fold for Margin {
+    type Output = Margin;
+
+    fn fold(self, outer: Self::Output) -> Self::Output {
+        let sides =
+            self.sides
+                .zip(outer.sides)
+                .map(|(inner, outer)| match (inner, outer) {
+                    (Some(value), Some(outer)) => Some(value.fold(outer)),
+                    (Some(value), None) => Some(value),
+                    (None, Some(outer)) => Some(outer),
+                    (None, None) => None,
```

Comment:
Simplify
```suggestion
                    (Some(value), Some(outer)) => Some(value.fold(outer)),
                    _ => inner.or(outer),
```

Reply:
(If Clippy complains about this, try to use `if let ... else` instead of a `match`)

Reply:
Done.